.github/workflows/ci.yml

```diff
@@ -76,6 +77,16 @@ jobs:
         if: always()
         uses: ./.github/actions/upload-heapdumps
 
+      - name: Surefire reports
+        if:  (!cancelled() && contains(fromJSON(${{ env.RELEASE_BRANCHES }}), github.ref) && contains(fromJSON('["push", "workflow_dispatch"]'), github.event_name))
+        uses: ./.github/actions/surefire-reports
+        with:
+          testrun-name: surefire-unit-tests
+          reports-path: |
```

Comment:
Wonder if we could just specify a pattern that includes all surefire reports instead of having to specify it for each individual job?

Also, could you use "job-name" instead of "testrun-name" to make it consistent with upload-flaky-tests?

As a follow-up to this PR I would probably want to refactor upload-headdumps, upload-flaky-test, and this one to a single action upload-reports or something so we don't have to have so much boilerplate.